### PR TITLE
Feature/add collection search extension V2

### PR DESCRIPTION
**Pull request by vincentsarago**
simpler version of #681 

This PR doesn't add the`/collections - POST` endpoint because:
- it's not mandatory in the spec 
- it will conflict with the transaction extension 😬 

We could in another PR show (in the documentation) how to create a `/collections - POST`  which support both `search and transaction`

cc @tjellicoe-tpzuk

### CODE REVIEW

File: stac_fastapi/extensions/stac_fastapi/extensions/core/collection_search/request.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    bbox: Optional[BBox] = attr.ib(default=None, converter=_bbox_converter)
+    datetime: Optional[DateTimeType] = attr.ib(
+        default=None, converter=_datetime_converter
+    )
+    limit: Annotated[
+        Optional[int],
+        Query(
+            description="Limits the number of results that are included in each page of the response."  # noqa: E501
+        ),
+    ] = attr.ib(default=None)
```
Comment (vincentsarago) [inline stac_fastapi/extensions/stac_fastapi/extensions/core/collection_search/request.py:27]: should we put a default to the limit ?

Comment (m-mohr) [reply stac_fastapi/extensions/stac_fastapi/extensions/core/collection_search/request.py:27]: The intention is to behave consistent with Item Search. What does Item Search put?

Comment (vincentsarago) [reply stac_fastapi/extensions/stac_fastapi/extensions/core/collection_search/request.py:27]: Item Search set `10` as default! https://github.com/stac-utils/stac-fastapi/blob/fe4d0dfb9812ec624b5af8bb10e32621b48e090a/stac_fastapi/types/stac_fastapi/types/search.py#L176-L181

**Comment by vincentsarago** [comment]
this is ready for review

File: CHANGES.md [hunk 2]
```diff
@@ -7,8 +7,8 @@
 * add more openapi metadata in input models [#734](https://github.com/stac-utils/stac-fastapi/pull/734)
 
 ### Added
-
-* Add Free-text Extension to third party extensions ([#655](https://github.com/stac-utils/stac-fastapi/pull/655))
+* Add Free-text Extension ([#655](https://github.com/stac-utils/stac-fastapi/pull/655))
+* Add Collection-Search Extension
```
Comment (jonhealy1) [inline CHANGES.md:11]: Add a link to the PR maybe

**Review by jonhealy1 — CHANGES_REQUESTED** [review]
Cool beans!

File: CHANGES.md (hunk 2 above)
Comment (vincentsarago) [inline CHANGES.md:11]:
```suggestion
* Add Collection-Search Extension ([#736](https://github.com/stac-utils/stac-fastapi/pull/736))
```

**Review by jonhealy1 — APPROVED** [review]

**Comment by m-mohr** [comment]
@jonhealy1 @vincentsarago I had another look at this and I'm wondering where the queryables enpoint is exposed. The filter extension, which is listed in the conformance classes, would require to be able to query based on queryables but I don't see where the link is added. Is that done in the backends?

**Comment by vincentsarago** [comment]
Yes that should be done by the users in their application.

**Comment by m-mohr** [comment]
@vincentsarago Why though? If you have the filter extension, you need the queryables link anyway. Could return an empty schema, but why this manual step?

Related PR: https://github.com/stac-utils/stac-fastapi-pgstac/pull/89

**Comment by m-mohr** [comment]
@vincentsarago I'm missing the queryables **link** (not the endpoint itself) in `GET /collections`, similar to how it's missing for global Item Search in `GET /` (see https://github.com/stac-utils/stac-fastapi-pgstac/pull/89 ). Maybe I'm missing something though?

**Comment by vincentsarago** [comment]
👍 the links are added in the client as you did in the stac-FastAPI-pgstac PR

**Pull request merged**